### DISCUSSION TITLE
Help me

### DISCUSSION THREAD

**Discussion by GoldTaro** [Huniko-Team/community#300] [Q&A] [answered]
Hi  Nice to meet you.
Now I'm trying to test on testnet.
This is my jettonWalletAddress : EQD7H6nQaN2PuLOckmZ1nFShJDubqOqLLK9RnEHEBFcD4JWN
And this is toAddress : 0QD5UUpPBopK-LayGy1N8U2JpqbYaSGGfcSDFhvt8qpY-1A2

const messageBody = beginCell()
            .storeUint(0x0f8a7ea5, 32)
            .storeUint(0, 64)
            .storeCoins(5001)
            .storeAddress(Address.parse(toAddress))
            .storeAddress(Address.parse(toAddress))
            .storeBit(0)
            .storeCoins(0)
            .storeBit(0)
            .endCell()

        const internalMessage = internal({
            to: jettonWalletAddress,
            value: toNano('1'),
            bounce: true,
            body: messageBody
        })

        const body = wallet.createTransfer({
            seqno,
            secretKey,
            messages: [internalMessage]
        })

        const externalMessage = external({
            to: address,
            init: neededInit,
            body
        })

        const externalMessageCell = beginCell().store(storeMessage(externalMessage)).endCell()

        const signedTransaction = externalMessageCell.toBoc()
        const hash = externalMessageCell.hash().toString('hex')

        console.log('hash:', hash)

        await client.sendFile(signedTransaction)
And I'm trying to send test 1 USDT.

This is my test USDT Master Address : kQAau4V6DsCRy_fmfDymJO5H_kLnIkjjKYLkXbBM0PZ1rtZy

I don't understand the difference between jetton address and wallet address.

It seems like difference between wallet address and token account address in solana.

Please let me know what did I make mistake in my code.

**Comment by OmO-sketch** [comment, marked as answer]
const messageBody = beginCell()
.storeUint(0x0f8a7ea5, 32)
.storeUint(0, 64)
.storeCoins(5001)                                                         //Enter Jetton value instead of 5001.
.storeAddress(Address.parse(toAddress))
.storeAddress(Address.parse(toAddress))
.storeBit(0)
.storeCoins(0)
.storeBit(0)
.endCell()

    const internalMessage = internal({
        to: jettonWalletAddress,
        value: toNano('1'),                                               //The amount of fee for sending Jetton
        bounce: true,
        body: messageBody
    })

    const body = wallet.createTransfer({
        seqno,
        secretKey,
        messages: [internalMessage]
    })

    const externalMessage = external({
        to: address,
        init: neededInit,
        body
    })

    const externalMessageCell = beginCell().store(storeMessage(externalMessage)).endCell()

    const signedTransaction = externalMessageCell.toBoc()
    const hash = externalMessageCell.hash().toString('hex')

    console.log('hash:', hash)

    await client.sendFile(signedTransaction)
